npm_nodejs/compatibility_mode.md

```diff
@@ -1,31 +1,108 @@
 ## Node.js compability mode
 
-Starting with v1.15 Deno provides Node compatiblity mode that can be activated
-by passing `--compat` flag in CLI.
-
-> ⚠️ Currently using compatiblity mode requires the `--unstable` flag, and if
-> you are loading CJS modules, the `--allow-read` flag is needed too.
-
-When using compatibility mode there are a few things happening behind the
-scenes:
-
-- Node globals are available in the global scope, so you can access `process`,
-  `global`, `Buffer`, `setImmediate`, `clearImmediate`. This is done by
-  executing
+Starting with v1.15 Deno provides Node compatiblity mode that makes it possible
+to run a subset of programs authored for Node.js directly in Deno. Compatiblity
+mode can be activated by passing `--compat` flag in CLI.
+
+> ⚠️ Using compatiblity mode currently requires the `--unstable` flag. If you
+> intend to use CJS modules, the `--allow-read` flag is needed as well.
+
+> ⚠️ Package management is currently out of scope for Node.js compatiblity mode.
+> For the time being we suggest to keep using your current solution (`npm`,
+> `yarn`, `pnpm`).
+
+### Example
+
+[`eslint`](https://eslint.org/) is a very popular tool used by most of Node.js
+projects. Let's run `eslint` using Deno in Node.js compatiblity mode. Assuming
+that `eslint` is already installed locally (either using `npm install eslint` or
+`yarn install eslint`) we can do so like:
+
+```sh
+$ ls
+.eslintrc.json
+node_modules
+package.json
+test.js
+test.ts
+$ cat test.js
+function foo() {}
+
+$ cat test.ts
+function bar(): any {}
+
+$ deno run \
+  --compat --unstable \
+  --allow-read --allow-write=./ --allow-env \
+  node_modules/eslint/bin/eslint.js test.js test.ts
+
+/dev/test.js
+  1:10  warning  'foo' is defined but never used  @typescript-eslint/no-unused-vars
+  1:16  error    Unexpected empty function 'foo'  @typescript-eslint/no-empty-function
+
+/dev/test.ts
+  1:10  warning  'bar' is defined but never used           @typescript-eslint/no-unused-vars
+  1:17  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
+  1:21  error    Unexpected empty function 'bar'           @typescript-eslint/no-empty-function
+
+✖ 5 problems (2 errors, 3 warnings)
+```
+
+> ⚠️ Notice that ESLint is run with limited set of permissions. We only give it
+> access to the read from the file system, write to current directory and access
+> environmental variables. Programs run in compatility mode are subject to
+> Deno's permission model.
```

Comment:
This is a great advantage of using deno compat mode over Node.js itself 👍

Reply:
Indeed!